src/STLDocument.cs

```diff
@@ -149,7 +134,19 @@ public static bool IsText(Stream stream)
         /// <returns>True if the <see cref="STLDocument"/> is binary-based, otherwise false.</returns>
         public static bool IsBinary(Stream stream)
         {
-            return !IsText(stream);
+            if (stream == null) throw new NullReferenceException(nameof(stream));
+
+            using var reader = new BinaryReader(stream, Encoding.Default, true);
```

Comment:
Can this be `const`? I haven't been in C#-world in a while, so not sure. 

```suggestion
            using const reader = new BinaryReader(stream, Encoding.Default, true);
```

Reply:
Or perhaps it should just be `using` rather than `using var` -- I assume a `using` variable is always constant.